tests/boards/intel_adsp/hda_log/src/logger.c

```diff
@@ -0,0 +1,67 @@
+/* Copyright (c) 2022 Intel Corporation
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log_backend_cavs_hda.h>
+#include <logging/log_backend.h>
+#include <logging/log_ctrl.h>
+#include <kernel.h>
+#include <string.h>
+#include <ztest.h>
+#include <cavs_ipc.h>
+#include "tests.h"
+
+#define CHANNEL 6
+#define HOST_BUF_SIZE 512
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(hda_test, LOG_LEVEL_DBG);
+
+#define IPC_TIMEOUT K_MSEC(500)
+
+void hda_log_hook(uint32_t written)
+{
+	/* We *must* send this, but we may be in a timer ISR, so we are
+	 * forced into a retry loop without timeouts and such.
+	 */
+	bool done = false;
+
+	/* Previous message may not be done yet, wait for that */
+	do {
+		done = cavs_ipc_is_complete(CAVS_HOST_DEV);
+	} while (!done);
+
+	/*  Now send the next one */
+	do {
+		done = cavs_ipc_send_message(CAVS_HOST_DEV, IPCCMD_HDA_PRINT,
+					     (written << 8) | CHANNEL);
+	} while (!done);
+}
+
+
+void test_hda_logger(void)
+{
```

Comment:
Test coverage pedantry: Not seeing a test case that exercises watermark flush or the overflow conditions.